nupic/research/monitor_mixin/plot.py

```diff
@@ -46,40 +49,70 @@ def __init__(self, monitor, title):
     plt.show()
 
 
+
+  def _initFigure(self):
+    fig = plt.figure()
+    fig.suptitle(self._prettyPrintTitle())
+    return fig
+
+
+
+  def _prettyPrintTitle(self):
+    if self._monitor.mmName is not None:
+      return "[{0}] {1}".format(self._monitor.mmName, self._title)
+    return self._title
+
+
+
   def addGraph(self, data, position=111, xlabel=None, ylabel=None):
     ax = self._addBase(position, xlabel=xlabel, ylabel=ylabel)
     ax.plot(data)
     plt.draw()
 
 
-  def addHistogram(self, data, position=111, xlabel=None, ylabel=None,
-                   bins=None):
+
+  def addImage(self, data, position=111, xlabel=None, ylabel=None,
+               cmap=colorModel.Greys, aspect="auto", interpolation="nearest"):
     """
-    @param bucketSize (int) Size of each bucket
+    Adds an image to the plot's figure.
+    :param data: a 2D array
+    :param position: A 3-digit number. The first two digits define a 2D grid
+            where subplots may be added. The final digit specifies the nth grid
+            location for the added subplot
+    :param xlabel: text to be displayed on the x-axis
+    :param ylabel: text to be displayed on the y-axis
+    :param cmap: color map used in the rendering
+    :param aspect: how aspect ratio is handled during resize
+    :param interpolation: interpolation method
     """
+    if data is not None:
```

Comment:
Is this check necessary? It would be simpler to just assume `data` is not `None`, and let it throw an error if that condition is violated.